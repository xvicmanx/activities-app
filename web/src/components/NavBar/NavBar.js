import React from 'react';
import { Route, Switch, Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { Image } from 'react-bulma-components';

import { logOutUser } from '../../entities/users/redux/UsersActions';

import menuIcon from "../../assets/images/menu.svg";

import './NavBar.css';

const NavBar = ({ inverted, right }) => {
  const { Users } = useSelector(state => state);
  const dispatch = useDispatch();
  const additionalClasses = [];

  if (inverted) {
    additionalClasses.push('inverted');
  }

  if (right) {
    additionalClasses.push('right');
  }

  if (!Users.data) {
    return (
      <div className={`NavBar ${additionalClasses.join(' ')}`}>
        <Switch>
          <Route exact path="/login">
            <Link className="NavBar__link" to="/">
              Atras
            </Link>
          </Route>
        </Switch>
      </div>
    );
  }

  return (
    <div className={`NavBar ${additionalClasses.join(' ')}`}>
      <Link className="NavBar__link menu" to="/home/display-menu">
        <Image
          src={menuIcon}
          alt="menu"
        />
      </Link>

      <Link
        onClick={() => dispatch(logOutUser())}
        className="NavBar__link"
        to="/"
      >
        Cerrar Sesion
      </Link>
    </div>
  );
};

export default NavBar;
