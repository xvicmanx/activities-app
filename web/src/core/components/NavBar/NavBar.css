.NavBar {
  background-color: #5caaf6;
  padding: 15px;
  margin-bottom: 20px;
}

.NavBar__link {
  margin: 0 15px;
  color: #fff;
}


.NavBar.inverted {
  background: #fff;
  border-bottom: 1px solid #f3f3f3;
  margin-bottom: 0;
}

.NavBar.right {
  display: flex;
  justify-content: flex-end;
}

.NavBar.inverted .NavBar__link{
  color: #76858e;
}

.NavBar__link.menu {
  position: absolute;
  left: 10px;
}

.NavBar__link.menu .image {
  border: 1px solid;
  padding: 5px;
  border-radius: 5px;
  width: 2rem;
  margin-top: -4px;
}

@media only screen and (min-width: 1024px) {
  .NavBar__link.menu {
    display: none;
  }
}