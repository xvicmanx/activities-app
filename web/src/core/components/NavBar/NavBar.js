// @flow

import React from 'react';
import { Link } from 'react-router-dom';
import { Image } from 'react-bulma-components';

import menuIcon from '../../../assets/images/menu.svg';

import './NavBar.css';

type Props = {
  onLogoutClick?: Function,
};

const NavBar = ({ onLogoutClick }: Props): React$Element<any> => (
  <div className="NavBar right">
    <Link className="NavBar__link menu" to="/home/display-menu">
      <Image src={menuIcon} alt="menu" />
    </Link>

    <Link onClick={onLogoutClick} className="NavBar__link" to="/">
      Cerrar Sesion
    </Link>
  </div>
);

export default NavBar;
