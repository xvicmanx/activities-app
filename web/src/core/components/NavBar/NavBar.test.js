import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import NavBar from './NavBar';

describe('NavBar', () => {
  it('renders properly', () => {
    const renderer = new ShallowRenderer();
    const onClick = jest.fn();
    const result = renderer.render(<NavBar onLogoutClick={onClick} />);
    expect(result).toMatchSnapshot();
  });
});
