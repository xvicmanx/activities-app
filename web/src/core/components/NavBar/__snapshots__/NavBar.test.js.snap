// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders properly 1`] = `
<div
  className="NavBar right"
>
  <ForwardRef
    className="NavBar__link menu"
    to="/home/display-menu"
  >
    <Image
      alt="menu"
      fallback="https://bulma.io/images/placeholders/480x480.png"
      rounded={false}
      src="menu.svg"
    />
  </ForwardRef>
  <ForwardRef
    className="NavBar__link"
    onClick={[MockFunction]}
    to="/"
  >
    Cerrar Sesion
  </ForwardRef>
</div>
`;
